airflow/utils/helpers.py

```diff
@@ -213,4 +213,4 @@ def build_airflow_url_with_query(query: Dict[str, Any]) -> str:
     'http://0.0.0.0:8000/base/graph?dag_id=my-task&root=&execution_date=2020-10-27T10%3A59%3A25.615587
     """
     view = conf.get('webserver', 'dag_default_view').lower()
-    return f"/{view}?{parse.urlencode(query)}"
+    return url_for(f'Airflow.{view}', **query)
```

Comment:
Kwrags of `url_for` are by default used for endpoint variable arguments (e.g. `/foo/{var}`), and only unrecognised key-values are appended as a query string. This works now since (I think) none of Airflow’s endpoints currently take any arguments, but could introduce subtle bugs in the future. Something like this is probably better for maintenance down the road.

```python
url = url_for(f"Airflow.{view}")
f"{url}?{urllib.parse.urlencode(query)}"
```

Reply:
Great insight! I will follow your suggestion!